cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskManager.java

```diff
@@ -131,54 +101,30 @@ public synchronized int leadershipMovementConcurrency() {
-    dropwizardMetricRegistry.register(MetricRegistry.name(metricName, GAUGE_REPLICA_ACTION_PENDING),
-                                      (Gauge<Integer>) _executionTaskTracker::numPendingReplicaAction);
-    dropwizardMetricRegistry.register(MetricRegistry.name(metricName, GAUGE_LEADERSHIP_ACTION_PENDING),
-                                      (Gauge<Integer>) _executionTaskTracker::numPendingLeadershipAction);
-    dropwizardMetricRegistry.register(MetricRegistry.name(metricName, GAUGE_REPLICA_ACTION_ABORTING),
-                                      (Gauge<Integer>) _executionTaskTracker::numAbortingReplicaAction);
-    dropwizardMetricRegistry.register(MetricRegistry.name(metricName, GAUGE_LEADERSHIP_ACTION_ABORTING),
-                                      (Gauge<Integer>) _executionTaskTracker::numAbortingLeadershipAction);
-    dropwizardMetricRegistry.register(MetricRegistry.name(metricName, GAUGE_REPLICA_ACTION_ABORTED),
-                                      (Gauge<Integer>) _executionTaskTracker::numAbortedReplicaAction);
-    dropwizardMetricRegistry.register(MetricRegistry.name(metricName, GAUGE_LEADERSHIP_ACTION_ABORTED),
-                                      (Gauge<Integer>) _executionTaskTracker::numAbortedLeadershipAction);
-    dropwizardMetricRegistry.register(MetricRegistry.name(metricName, GAUGE_REPLICA_ACTION_DEAD),
-                                      (Gauge<Integer>) _executionTaskTracker::numDeadReplicaAction);
-    dropwizardMetricRegistry.register(MetricRegistry.name(metricName, GAUGE_LEADERSHIP_ACTION_DEAD),
-                                      (Gauge<Integer>) _executionTaskTracker::numDeadLeadershipAction);
-    dropwizardMetricRegistry.register(MetricRegistry.name(metricName, GAUGE_ONGOING_EXECUTION_IN_KAFKA_ASSIGNER_MODE),
-                                      (Gauge<Integer>) _executionTaskTracker::isOngoingExecutionInKafkaAssignerMode);
-    dropwizardMetricRegistry.register(MetricRegistry.name(metricName, GAUGE_ONGOING_EXECUTION_IN_NON_KAFKA_ASSIGNER_MODE),
-                                      (Gauge<Integer>) _executionTaskTracker::isOngoingExecutionInNonKafkaAssignerMode);
-  }
-
   /**
    * Returns a list of execution proposal that moves the partitions.
    */
   public synchronized List<ExecutionTask> getReplicaMovementTasks() {
-    Map<Integer, Integer> readyBrokers = new HashMap<>();
-    int partitionMovementConcurrency = partitionMovementConcurrency();
-    for (Map.Entry<Integer, Integer> entry : _inProgressReplicaMovementsByBrokerId.entrySet()) {
+    Map<Integer, Integer> readyBrokers = getReadyBrokers(_inProgressReplicaMovementsByBrokerId,
+                                                         partitionMovementConcurrency());
+    return _executionTaskPlanner.getReplicaMovementTasks(readyBrokers, _inProgressPartitions);
+
+  }
+
+  private Map<Integer, Integer> getReadyBrokers(Map<Integer, Integer> inProgressReplicaMovementsByBrokerId,
```

Comment:
The reason we factor out this chunk of code is because in 2.0 branch, there will be 2 counterpart method for `getReplicaMovementTasks()`, one for inter-broker movement and one for intra-broker movement,  and this method will be the common logic in two method.  Here I want to make two branches' code structure similar.